### PR TITLE
Fix link to SciPy Code of Conduct in Numpy Code of Conduct

### PR DESCRIPTION
**Pull request by psobolewskiPhD**
Fixes gh-595

Using the link provided on the SciPy Community page: 
https://scipy.org/community/

### CODE REVIEW

File: content/en/code-of-conduct.md [hunk 1]
```diff
@@ -80,4 +80,4 @@ The Committee will respond to any report as soon as possible, and at most within
 
 We are thankful to the groups behind the following documents, from which we drew content and inspiration:
 
-- [The SciPy Code of Conduct](https://docs.scipy.org/doc/scipy/reference/dev/conduct/code_of_conduct.html)
+- [The SciPy Code of Conduct](https://scipy.github.io/devdocs/dev/conduct/code_of_conduct.html)
```
Comment (seberg) [inline content/en/code-of-conduct.md:83]: The github.io site doesn't seem like the best solution to me?  That seems more of a deployment detail.  Maybe rather: https://docs.scipy.org/doc/scipy/dev/conduct/code_of_conduct.html maybe the "reference" part of the URL went away at some point.

Comment (psobolewskiPhD) [reply content/en/code-of-conduct.md:83]: 🤷‍♂️ 
I used the link that SciPy uses on it's website, figuring that was the authoritative source.
Happy to change to this version though, it seems to work.

Comment (rgommers) [reply content/en/code-of-conduct.md:83]: It looks like https://scipy.org/community/ and the footer on that site use https://docs.scipy.org/doc/scipy/dev/conduct/code_of_conduct.html consistently. So +1 for using that link.

Comment (psobolewskiPhD) [reply content/en/code-of-conduct.md:83]: OK, I will go with that, but let the record reflect that SciPy started using that URL today:
https://github.com/scipy/scipy/pull/16963/files
😉

Comment (rgommers) [reply content/en/code-of-conduct.md]: Ah, I did not realize that.

**Review by rgommers — APPROVED** [review]
Thanks @psobolewskiPhD. And thanks for the review @seberg

**Comment by InessaPawson** [comment]
Hi-five on merging your first pull request to NumPy, @psobolewskiPhD! We hope you stick around! Your choices aren’t limited to writing code – you can review pull requests, help us stay on top of new and old  issues, develop educational material, work on our website, add or improve graphic design, create marketing materials, translate website content, write grant proposals, and help with other fundraising initiatives. For more info, check out: https://numpy.org/contribute

Also, consider joining our mailing list. This is a great way to connect with other cool people in our community and be part of important conversations that affect the development of NumPy: https://mail.python.org/mailman/listinfo/numpy-discussion

**Pull request merged**